invenio_users_resources/services/users/service.py

```diff
@@ -145,11 +146,39 @@ def block(self, identity, id_, uow=None):
         user.commit()
 
         uow.register(RecordIndexOp(user, indexer=self.indexer, index_refresh=True))
+
+        # Start a task to execute after actions asynchronously
+        execute_moderation_actions.delay(user.id, "block")
+
+        return True
+
+    @unit_of_work()
+    def restore(self, identity, id_, uow=None):
+        """Restores a user."""
+        user = UserAggregate.get_record(id_)
+        if user is None:
+            # return 403 even on empty resource due to security implications
+            raise PermissionDeniedError()
+
+        self.require_permission(identity, "manage", record=user)
+
+        user.model.active = True
+        user.model.blocked_at = None
+        # TODO if we restore the user, should they be verified?
```

Comment:
question: when we restore a user, should the user be verified? I see this as a transition from BLOCKED to APPROVED

Reply:
Discussed IRL: the user should be verified when restored, as we do on Zenodo